core/creators/themedui.js

```diff
@@ -384,6 +388,62 @@ CKEDITOR.replaceClass = 'ckeditor';
 		return editor;
 	}
 
+	// Delay editor creation if given element is detached from DOM.
+	//
+	// Require `config.delayIfDetached` to be `true`.
+	//
+	// If `config.delayIfDetached_callback` is declared as function, it will be invoked with single argument:
+	// function, that should be called to create editor.
+	//
+	// Otherwise: periodically (with `setInterval`) run check if element is attached to DOM and create editor automatically.
+	//
+	// ```js
+	//	CKEDITOR.replace( detachedEditorElement, {
+	//		delayIfDetached: true,
+	//		delayIfDetached_interval: 50,
+	//		delayIfDetached_callback:RegisterCallback
+	//		}
+	//	} );
+	// ```
+	//
+	// @param {CKEDITOR.element} element The DOM element.
+	// @param {String} mode CKEditor mode.
+	// @param {Object} [config] The specific configuration to apply to this
+	// editor instance. Configuration set here will override the global CKEditor settings.
+	// @param {String} [data] Initial value for the instance.
+	function delayCreationOnDetachedElement( element, mode, config, data ) {
+		if ( !config || !config.delayIfDetached || !element.isDetached() ) {
+			return false;
+		}
+
+		if ( !config.delayIfDetached_callback ) {
+			CKEDITOR.config.delayIfDetached_callback( intervalyAttemptInstanceCreation );
+			return true;
+		}
+
+		config.delayIfDetached_callback( function() {
+			createInstance( element, config, data, mode );
+		} );
+
+		return true;
+
+		function intervalyAttemptInstanceCreation() {
+			var delay = parseFloat( config.delay );
+			delay = isNaN( delay ) ? CKEDITOR.config.delayIfDetached_interval : delay;
+
+			CKEDITOR.warn( 'Editor creation will be delayed. We will keep trying every ' + delay + 'ms until target element will be reattached to DOM.' );
+
+			var intervalId = setInterval( function() {
+				if ( !element.isDetached() ) {
+					clearInterval( intervalId );
+
+					CKEDITOR.warn( 'Editor instance will be created now. Target element is attached to DOM.' );
```

Comment:
As mentioned in https://github.com/ckeditor/ckeditor4/pull/4463/files#r590134485, this should be extracted as a defined error message.